dipy/tracking/vox2track.pyx

```diff
@@ -162,6 +161,172 @@ def streamline_mapping(streamlines, voxel_size=None, affine=None,
+            next_pt[dim_idx] = streamline[point_idx + 1, dim_idx]
+            direction[dim_idx] = next_pt[dim_idx] - current_pt[dim_idx]
+            current_edge[dim_idx] = current_pt[dim_idx]
+
+        # Compute norm
+        direction_norm = norm(direction[0], direction[1], direction[2])
+
+        # Set the "dist" var to compute remaining length of vector
+        # to process
+        remaining_distance = direction_norm
+
+        # Check if it's already a real edge. If not, find the closest edge.
+        if floor(current_edge[0]) != current_edge[0] and \
+           floor(current_edge[1]) != current_edge[1] and \
+           floor(current_edge[2]) != current_edge[2]:
+            # All coordinates are not "integers", and therefore, not on the
+            # edge. Fetch the closest edge.
+            c_get_closest_edge(current_pt[0], current_pt[1], current_pt[2],
+                               direction[0], direction[1], direction[2],
+                               current_edge)
+
+        # TODO Could condition be optimized?
+        while True:
+            # Compute the smallest ratio of direction's length to get to an
+            # edge. This effectively means we find the first edge
+            # encountered
+            # Set large value for length_ratio
+            length_ratio = 10000
+            for dim_idx in range(3):
+                if direction[dim_idx] != 0:
+                    length_ratio = fmin(
+                        fabs((current_edge[dim_idx] - current_pt[dim_idx])
+                            / direction[dim_idx]),
+                        length_ratio)
+
+            # Check if last point is already on an edge
+            remaining_distance -= length_ratio * direction_norm
+            if remaining_distance < 0 and not fabs(remaining_distance) < 1e-8:
+                break
+
+            # Find the coordinates of voxel containing current point, to
+            # tag it in the map
+            half_ratio = 0.5 * length_ratio
+            x = <int>(current_pt[0] + half_ratio * direction[0])
+            y = <int>(current_pt[1] + half_ratio * direction[1])
+            z = <int>(current_pt[2] + half_ratio * direction[2])
+            if mask[x, y, z]:
+                return 1
+
+            # current_pt is moved to the closest edge
+            for dim_idx in range(3):
+                current_pt[dim_idx] = \
+                    length_ratio * direction[dim_idx] + current_pt[dim_idx]
+
+                # Snap really small values to 0.
+                if fabs(current_pt[dim_idx]) <= 1e-16:
+                    current_pt[dim_idx] = 0.0
+
+            c_get_closest_edge(current_pt[0], current_pt[1], current_pt[2],
+                               direction[0], direction[1], direction[2],
+                               current_edge)
+
+    # Check last point
+    x = <int>next_pt[0]
+    y = <int>next_pt[1]
+    z = <int>next_pt[2]
+    return mask[x, y, z]
+
+
+@cython.boundscheck(False)
```

Comment:
It seems those 2 lines were added on master in the meantime so I'll ignore this comment.